tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/AsyncMethodInvoker_InvokeBegin_Integration.cs

```diff
@@ -27,73 +29,24 @@ namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Wcf
     public class AsyncMethodInvoker_InvokeBegin_Integration
     {
-        /// <summary>
-        /// OnMethodBegin callback
-        /// </summary>
-        /// <typeparam name="TTarget">Type of the target</typeparam>
-        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
-        /// <param name="instanceArg">RequestContext instance</param>
-        /// <param name="inputs">Input arguments</param>
-        /// <param name="callback">Callback argument</param>
-        /// <param name="state">State argument</param>
-        /// <returns>Calltarget state value</returns>
-        internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, object instanceArg, object[] inputs, AsyncCallback callback, object state)
-        {
-            // TODO Just use the OperationContext.Current object to get the span information
-            // context.IncomingMessageHeaders contains:
-            //  - Action
-            //  - To
-            //
-            // context.IncomingMessageProperties contains:
-            // - ["httpRequest"] key to find distributed tracing headers
-            if (!Tracer.Instance.Settings.IsIntegrationEnabled(WcfCommon.IntegrationId) || !Tracer.Instance.Settings.DelayWcfInstrumentationEnabled || WcfCommon.GetCurrentOperationContext is null)
-            {
-                return CallTargetState.GetDefault();
-            }
-
-            var operationContext = WcfCommon.GetCurrentOperationContext?.Invoke();
-
-            if (operationContext != null && operationContext.TryDuckCast<IOperationContextStruct>(out var operationContextProxy))
-            {
-                var requestContext = operationContextProxy.RequestContext;
-
-                // First, capture the active scope
-                var activeScope = Tracer.Instance.InternalActiveScope;
-                var spanContextRaw = DistributedTracer.Instance.GetSpanContextRaw() ?? activeScope?.Span?.Context;
-
-                var useWcfWebHttpResourceNames = Tracer.Instance.Settings.WcfWebHttpResourceNamesEnabled;
-
-                // Then, create the new scope
-                var scope = WcfCommon.CreateScope(requestContext, useWcfWebHttpResourceNames);
-
-                if (scope != null)
-                {
-                    // Save the scope to retrieve it during the AsyncMethodInvoker.InvokeEnd method
-                    WcfCommon.Scopes.Add(((IDuckType)requestContext).Instance, scope);
-                }
-
-                return new CallTargetState(scope, activeScope, spanContextRaw);
-            }
-            else
-            {
-                return CallTargetState.GetDefault();
-            }
-        }
-
         internal static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, in CallTargetState state)
         {
-            if (exception != null)
+            if (exception is not null)
```

Comment:
Note: The new AfterReceiveRequestIntegration and BeforeSendReplyIntegration integrations start/stop the span. This will only decorate the existing span with exception information.